src/main/java/ch/njol/skript/effects/EffToggle.java

```diff
@@ -71,15 +78,17 @@
 		setDataMethod = mh;
 	}
 	
-	@SuppressWarnings("null")
-	private Expression<Block> blocks;
+	private Expression<?> toggledExpr;
 	private int toggle;
 	
-	@SuppressWarnings({"unchecked", "null"})
 	@Override
 	public boolean init(final Expression<?>[] vars, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		blocks = (Expression<Block>) vars[0];
+		toggledExpr = (Expression<?>) vars[0];
 		toggle = matchedPattern - 1;
+		if (toggledExpr.getReturnType().isAssignableFrom(Boolean.class) && ChangerUtils.acceptsChange(toggledExpr, ChangeMode.SET)) {
```

Comment:
```suggestion
		if (toggledExpr.getReturnType() == Boolean.class && !ChangerUtils.acceptsChange(toggledExpr, ChangeMode.SET, Boolean.class)) {
```
I was wrong about the return type check, it should only check if it definitely returns a boolean